File: app/src/main/java/com/cleanup/todoc/domaine/usecase/DeleteTaskUseCase.java
package com.cleanup.todoc.domaine.usecase;

import com.cleanup.todoc.datasource.model.TaskEntity;
import com.cleanup.todoc.data.repository.TaskRepository;
import com.cleanup.todoc.domaine.model.TaskDomain;

public class DeleteTaskUseCase {
    private final TaskRepository repository;
    public DeleteTaskUseCase(TaskRepository repository){this.repository=repository;}
    public void deleteTask(TaskDomain taskDomain){
        repository.delete(taskDomain);

}
}

File: app/src/main/java/com/cleanup/todoc/domaine/usecase/DeleteTaskUseCase.kt
package com.cleanup.todoc.domaine.usecase;

import androidx.lifecycle.Transformations;

import com.cleanup.todoc.data.repository.TaskRepository;
import com.cleanup.todoc.domaine.TaskDomainUiMapper;
import com.cleanup.todoc.domaine.model.TaskDomain;
import com.cleanup.todoc.presentation.model.Task;

public class DeleteTaskUseCase {
    private final TaskRepository repository;
    public DeleteTaskUseCase(TaskRepository repository){this.repository=repository;}
    public void deleteTask(Task task){
        repository.delete(TaskDomainUiMapper.mapToDomain(task));
    }
}
